notebook2/notebook/notebook.py

```diff
@@ -315,6 +347,51 @@ def wait_websocket(ws):
     log.info(f'notification sent to user for {svc_name} {pod_name}')
 
 
+@app.route('/auth0-callback')
+def auth0_callback():
+    # https://github.com/auth0-samples/auth0-python-web-app/commit/d048d6497caa714c52e8411a5f37500787e37305
+    auth0.authorize_access_token()
+
+    userinfo = auth0.get('userinfo').json()
+
+    email = userinfo['email']
+    workshop_password = session['workshop_password']
+    del session['workshop_password']
+
+    if AUTHORIZED_USERS.get(email) is None and workshop_password != PASSWORD:
+        return redirect(flask.url_for('login_page', unauthorized = True))
```

Comment:
It seems a bit odd to pun the login page as both a login page and an unauthorized page. I realize there's some HTML duplication, but, code-structure-wise, it seems much clearer to just have a `/unauthorized` page. What do you think?

Reply:
I saw it as an action that originated on the login page, and updates that page’s state after completion (so login page manages both submission and handles the response).

I can see your point, happy to create an unauthorized page.

Reply:
Done, /error endpoint added